### PR TITLE
fix publisher form: aliases -> otherNames, earlierName -> previousNames

### PR DESCRIPTION
**Pull request by mvillivald**
- For now, "aliases" are sent as an array named "otherNames". "earlierName" accordingly was renamed to "previousNames" and is also sent as an array.
- I have slightly fixed the translations as well.

### CODE REVIEW

File: src/frontend/components/form/publisherRegistrationForm/utils.js [hunk 1]
```diff
@@ -33,14 +33,23 @@ import {Typography} from '@mui/material';
 export function formatPublisher(values) {
   const newPublisherCategory = values.publisherCategory && values.publisherCategory.value;
   const newClassification = values.classification.map(item => item.value.toString());
+  const newOtherNames = values.otherNames.split('\n');
```
Comment (aatuny) [inline src/frontend/components/form/publisherRegistrationForm/utils.js:36]: Avoid implicit array formatting. In this case, it would be better to utilize component which formats and displays the values as they are interpreted. Follow similar pattern as with adding authors in isbn registration form.

File: src/frontend/components/form/publisherRegistrationForm/content.js [hunk 2]
```diff
@@ -43,15 +43,15 @@ export function getFormPages() {
           width: 'half'
         },
         {
-          name: 'aliases',
+          name: 'otherNames',
           type: 'textArea',
```
Comment (aatuny) [inline src/frontend/components/form/publisherRegistrationForm/content.js:47]: See other comment regarding value formatting. TextArea is not sufficient form field type for handling array values.

**Review by aatuny — CHANGES_REQUESTED** [review]

File: src/frontend/components/form/publisherRegistrationForm/PublisherNames.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
+import React from 'react';
```
Comment (aatuny) [inline src/frontend/components/form/publisherRegistrationForm/PublisherNames.js:1]: License header is missing.

File: src/frontend/components/form/publisherRegistrationForm/PublisherNames.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import {FormattedMessage} from 'react-intl';
```
Comment (aatuny) [inline src/frontend/components/form/publisherRegistrationForm/PublisherNames.js:2]: Instead of creating a custom component try utilizing Creatable component from react-select and add 'No other names' or similar text as default option. See if you can re-utilize component defined renderMultiSelect.js similar to what is defined in content.js for ISBN form author roles.

**Review by aatuny — CHANGES_REQUESTED** [review]

File: src/frontend/components/form/publisherRegistrationForm/utils.js [hunk 5]
```diff
@@ -33,14 +33,23 @@ import {Typography} from '@mui/material';
 export function formatPublisher(values) {
   const newPublisherCategory = values.publisherCategory && values.publisherCategory.value;
   const newClassification = values.classification.map(item => item.value.toString());
+  const newOtherNames = values.otherNames.map(item => item.value.toString());
```
Comment (aatuny) [inline src/frontend/components/form/publisherRegistrationForm/utils.js:36]: If no values are defined otherNames and previousNames results into undefined. You need to test whether values are defined and return empty array if no values are defined and mapped array if values are defined.

File: src/frontend/components/form/publisherRegistrationForm/utils.js [hunk 6]
```diff
@@ -33,14 +33,23 @@ import {Typography} from '@mui/material';
...
+  const newPreviousNames = values.previousNames.map(item => item.value.toString());
```
Comment (aatuny) [inline src/frontend/components/form/publisherRegistrationForm/utils.js:37]: Improve naming, use naming based on keys to simplify construction of new object.

File: src/frontend/components/form/publisherRegistrationForm/utils.js [hunk 7]
```diff
@@ -33,14 +33,23 @@ import {Typography} from '@mui/material';
...
   const {publicationDetails} = values;
   const {affiliateOf, affiliates, distributorOf, distributor, ...rest} = {...values};
 
   const newPublisher = {
```
Comment (aatuny) [inline src/frontend/components/form/publisherRegistrationForm/utils.js:41]: As you return the newly constructed object, there is no need to separately define the object - you can use object literal as return value.

**Review by aatuny — CHANGES_REQUESTED** [review]
Improve naming, fix bug which prevents sending form when no values are defined for otherNames or previousNames.

**Review by aatuny — APPROVED** [review]
